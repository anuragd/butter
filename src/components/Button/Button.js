import React, { Component } from 'react'
import PropTypes from 'prop-types'

import styles from './Button.less'

/**
 * Generic button component with animations
 * @version 0.0.1
 */
export default class Button extends Component {
  static propTypes = {
    /**
     * String that will appear inside the button as its label
     */
    label: PropTypes.string.isRequired,
    /**
     * Callback executed when mouse hovers over the component. Useful when tooltips or other related actions are needed
     */
    hoverHandler: PropTypes.func,
    /**
     * Callback for user click
     */
    clickHandler: PropTypes.func,
    /**
     * Setting this attribute will cause the button to be disabled. Disabled buttons will not fire any user interaction events
     */
    disabled: PropTypes.bool
  }

  constructor(props) {
    super(props)
    this.state = {
    };
    this.clickHandler = this.clickHandler.bind(this)
  }

  clickHandler(e) {
    if(this.props.clickHandler) this.props.clickHandler()
  }

  render() {
    const {
      label,
      disabled
    } = this.props

    return (
      <div className={styles.button_container}>
        <button className={disabled?styles.button_disabled:styles.button} onClick={this.clickHandler}>{label}</button>
      </div>
    )
  }
}
