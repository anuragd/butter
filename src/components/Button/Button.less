@import "../../styles/grid.less";
@import "../../styles/mixins.less";
@import "../../styles/animations.less";
@import "../../styles/type.less";

.button_container {
	padding: @base-size;
}

.button {
	height: @button-height;
	border-radius: @button-height;
	background: @white;
	min-width: @min-button-width;
	border: none;
	cursor: pointer;
	.regularFont();
	.highlightBorder();
	transition: @standard-transition;
	&:hover {
		background: @primary-bright;
		color: @white;
	}
	&:focus {
		.focusBox();
		background: @primary-bright;
		color: @white;
	}
}

.button_disabled {
	height: @button-height;
	border-radius: @button-height;
	background: @disable-text;
	min-width: @min-button-width;
	border: none;
	.regularFont();
	.noOutlines();
	color: @white;
	&:focus {
		.noOutlines();
	}
}