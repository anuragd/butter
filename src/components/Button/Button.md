Button example:

```jsx
<components.Button label="Button" />
```