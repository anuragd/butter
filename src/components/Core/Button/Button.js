import React, { Component } from 'react'
import PropTypes from 'prop-types'

import styles from './Button.less'

/**
 * Cross device compatible button. Should be positioned in the layout with a minimum space of 180px X 56px (determined by setting in `src/styles/grid.less`)
 * @version 0.0.1
 */
export default class Button extends Component {
  static propTypes = {
    /**
     * String that will appear inside the button as its label
     */
    label: PropTypes.string.isRequired,
    /**
     * Callback for user click. Mouse event is passed as a param.
     */
    onClick: PropTypes.func,
    /**
     * Setting this attribute will cause the button to be disabled. Disabled buttons will not fire any user interaction events
     */
    disabled: PropTypes.bool,

    /**
     * The `onMouseEnter` callback is fired when the mouse is moved over the button. Mouse event is passed as a param.
     */
    onMouseEnter: PropTypes.func,
    /**
     * The `onMouseLeave` callback is fired when the mouse is moved out of the buttons hit area. Mouse event is passed as a param.
     */
    onMouseLeave: PropTypes.func,
    /**
     * Callback for user focus event. Mouse event is passed as a param.
     */
    onFocus: PropTypes.func,
    /**
     * Callback for loss of focus from the component. Mouse event is passed as a param.
     */
    onBlur: PropTypes.func,
  }

  constructor(props) {
    super(props)
  }

  render() {
    const {
      label,
      disabled
    } = this.props

    return (
      <div className={styles.button_container}>
        <button 
          className={disabled?styles.button_disabled:styles.button} 
          onClick={this.props.onClick} 
          onMouseEnter={this.props.onMouseEnter}
          onMouseLeave={this.props.onMouseLeave}
          disabled={disabled}
          onFocus={this.props.onFocus}
          onBlur={this.props.onBlur}
          tabIndex="1"
          >
          {label}
        </button>
      </div>
    )
  }
}
