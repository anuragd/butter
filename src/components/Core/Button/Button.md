Normal button
```jsx noeditor
<Button label="Button" />
```
```jsx static
import { Button } from 'dcn-network-insights-ux';

...
...

render() {
    return (
        /* Rest of your markup */
        <Button 
            label="Button"
            onClick = {this.onClickHandler}
            onMouseEnter = {this.onMouseEnter}
            onMouseLeave = {this.onMouseLeave} />
    )
}

```

Disabled button
```jsx noeditor
<Button label="Button" disabled/>
```
```jsx static
import { Button } from 'dcn-network-insights-ux';

...
...

render() {
    return (
        /* Rest of your markup */
        <Button 
            label="Button"
            disabled/>
    )
}

```