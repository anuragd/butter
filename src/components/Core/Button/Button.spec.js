import React from 'react'
import renderer from 'react-test-renderer'

import Button from './Button'

test("Render a normal button", () => {
	const component = renderer.create(
		<Button label="Button" />
	)
	let tree = component.toJSON()
	expect(tree).toMatchSnapshot()
})
