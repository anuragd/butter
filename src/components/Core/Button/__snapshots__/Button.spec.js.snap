// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a normal button 1`] = `
<div
  className={undefined}
>
  <button
    className={undefined}
    disabled={undefined}
    onClick={undefined}
    onMouseEnter={undefined}
    onMouseLeave={undefined}
  >
    Button
  </button>
</div>
`;
